service.libraryautoupdate/edit_custom_paths.py

```diff
@@ -0,0 +1,72 @@
+import sys
+from kodi_six import xbmcgui
+import resources.lib.utils as utils
+from resources.lib.cronclasses import CustomPathFile
+dialog = xbmcgui.Dialog()
+
+#show the disclaimer - do this every time
+dialog.ok(utils.getString(30031),"",utils.getString(30032),utils.getString(30033))
+
+def selectPath(contentType):
+    path = {'expression':'0 */2 * * *','content':contentType}
+    
+    #select path to scan
+    path['path'] = dialog.browse(0,utils.getString(30023),contentType)
+
+    #create expression
+    if(path['path'] != ''):
+        path['expression'] = dialog.input(utils.getString(30056),path['expression'])
+    else:
+        #return nothing if dialog closed
+        return None
+    
+    return path
+
+def showMainScreen(contentType):
+    exitCondition = ""
+    customPaths = CustomPathFile(contentType)
+
+    while(exitCondition != -1):
```

Comment:
and not abortrequested...

Reply:
The edit_custom_paths.py file is launched from within the settings, not part of the service. It's a path editor to choose specific paths for updating. The exitCondition variable holds the user selection from a select dialog and then closes out when they hit cancel. abortrequested doesn't really work in this scenerio since it's not part of the service.